geodetic_utils/include/geodetic_utils/geodetic_conv.h

```diff
@@ -0,0 +1,217 @@
+    double Q = sqrt(1 + 2 * esq * esq * P);
+    double r_0 = -(P * esq * r) / (1 + Q)
+        + sqrt(
+            0.5 * a * a * (1 + 1.0 / Q) - P * (1 - esq) * z * z / (Q * (1 + Q)) - 0.5 * P * r * r);
+    double U = sqrt(pow((r - esq * r_0), 2) + z * z);
+    double V = sqrt(pow((r - esq * r_0), 2) + (1 - esq) * z * z);
+    double Z_0 = b * b * z / (a * V);
+    altitude = U * (1 - b * b / (a * V));
+    lat = rad2Deg(atan((z + e1sq * Z_0) / r));
+    lon = rad2Deg(atan2(y, x));
+  }
+
+  void ecef2Ned(const double x, const double y, const double z, double& north, double& east,
+                double& depth)
+  {
+    // Converts ECEF coordinate position into local-tangent-plane NED.
+    // Coordinates relative to given ECEF coordinate frame.
+
+    Eigen::Vector3d vect, ret;
+    vect(0) = x - _init_ecef_x;
+    vect(1) = y - _init_ecef_y;
+    vect(2) = z - _init_ecef_z;
+    ret = _ecef_to_ned_matrix * vect;
+    north = ret(0);
+    east = ret(1);
+    depth = -ret(2);
+  }
+
+  void ned2Ecef(const double north, const double east, const double depth, double& x, double& y,
+                double& z)
+  {
+    // NED (north/east/down) to ECEF coordinates
+    Eigen::Vector3d ned, ret;
+    ned(0) = north;
+    ned(1) = east;
+    ned(2) = -depth;
+    ret = _ned_to_ecef_matrix * ned;
+    x = ret(0) + _init_ecef_x;
+    y = ret(1) + _init_ecef_y;
+    z = ret(2) + _init_ecef_z;
+  }
+
+  void geodetic2Ned(const double lat, const double lon, const double altitude, double& north,
+                    double& east, double& depth)
+  {
+    // Geodetic position to local NED frame
+    double x, y, z;
+    geodetic2Ecef(lat, lon, altitude, x, y, z);
+    ecef2Ned(x, y, z, north, east, depth);
+  }
+
+  void ned2Geodetic(const double north, const double east, const double depth, double& lat,
+                    double& lon, double& altitude)
+  {
+    // Local NED position to geodetic coordinates
+    double x, y, z;
+    ned2Ecef(north, east, depth, x, y, z);
+    ecef2Geodetic(x, y, z, lat, lon, altitude);
+  }
+
+  void geodetic2Enu(const double lat, const double lon, const double altitude, double& east,
+                    double& north, double& up)
+  {
+    // Geodetic position to local ENU frame
+    double x, y, z;
+    geodetic2Ecef(lat, lon, altitude, x, y, z);
+
+    double aux_north, aux_east, aux_depth;
+    ecef2Ned(x, y, z, aux_north, aux_east, aux_depth);
+
+    east = aux_east;
+    north = aux_north;
+    up = -aux_depth;
+  }
+
+  void enu2Geodetic(const double east, const double north, const double up, double& lat,
+                    double& lon, double& altitude)
+  {
+    // Local ENU position to geodetic coordinates
+
+    const double aux_north = north;
+    const double aux_east = east;
+    const double aux_depth = -up;
+    double x, y, z;
+    ned2Ecef(aux_north, aux_east, aux_depth, x, y, z);
+    ecef2Geodetic(x, y, z, lat, lon, altitude);
+  }
+
+  double _init_lat;
+  double _init_lon;
+  double _init_h;
```

Comment:
why the preceding `_` here and above and I would prefer `initial_x`, `init` can always stand for initialize and initial.